src/features/query-bar/components/filter/Condition.js

```diff
@@ -113,14 +111,13 @@ const Condition = ({
     };
 
     const getColumnOptions = () => {
-        const { fieldId } = condition;
-        const column = columns && columns.find(c => c.id === fieldId);
+        const { columnDisplayText } = condition;
+        const column = columns && columns.find(c => c.displayName === columnDisplayText);
```

Comment:
Instead of using fieldId to check which column we're looking for, we can instead use column.displayName to identify the column.

Reply:
I'd prefer columnKey. "Display text" is a view concern and should not drive business logic.